### ISSUE TITLE
[java] UnusedLocalVariable: false-negative with array

### ISSUE THREAD

**Issue by Fernal73** [pmd/pmd#2130]
**Affects PMD Version:** 
6.19.0

**Rule:**
[UnusedLocalVariable](https://pmd.github.io/latest/pmd_rules_java_bestpractices.html#unusedlocalvariable)

**Description:**
An unused array object is not flagged as unused but flagged as DU-anomaly by rule DataflowAnomalyAnalysis. Both should happen. Comment in code states that this is a TODO.

https://github.com/pmd/pmd/blob/master/pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/bestpractices/UnusedLocalVariableRule.java
**Code Sample demonstrating the issue:**

```java
package pmdtests;

import java.lang.reflect.Constructor;

public final class Construct {

  private Construct() {
    throw new IllegalStateException("Private constructor.");
  }

  public static void main(String... args) {
    // get all visible constructors
    Constructor<?>[] constructors = String.class.getConstructors();
  }
}

```

**Running PMD through:** *[CLI | Ant | Maven | Gradle | Designer | Other]*

**Comment by adangel** [comment]
This has been fixed with PMD 7.0.0-rc1.
